### PR TITLE
feature: apply minimum continous range of child chain blocks to watcher

### PR DESCRIPTION
**Pull request by pgebal**

### CODE REVIEW

File: apps/omg_watcher/lib/block_getter/core.ex [hunk 1]
```diff
@@ -101,9 +99,9 @@ defmodule OMG.Watcher.BlockGetter.Core do
   @doc """
   Marks that childchain block `blknum` was processed
```
Comment (paulperegud) [inline apps/omg_watcher/lib/block_getter/core.ex]: Please update the doc

File: apps/omg_watcher/lib/block_getter/core.ex [hunk 2]
```diff
@@ -129,29 +127,29 @@ defmodule OMG.Watcher.BlockGetter.Core do
   end
 
   def get_blocks_to_consume(
-        %__MODULE__{blocks_to_consume: blocks} = state,
+        %__MODULE__{blocks_to_consume: blocks, block_interval: interval, last_consumed_block: last_consumed_block} =
```
Comment (paulperegud) [inline apps/omg_watcher/lib/block_getter/core.ex]: This function definitely deserves a doc :)

Comment (pgebal) [reply apps/omg_watcher/lib/block_getter/core.ex]: done

**Comment by paulperegud** [comment]
Some errors from dialyzer on 21.1 / 1.7.3.

```Elixir
apps/omg_api/lib/api.ex:27: The specification for 'Elixir.OMG.API':submit/1 states that the function might also return {'ok',#{'blknum':=integer(), 'tx_hash':=bitstring(), 'tx_index':=integer()}} but the inferred return is {'error','bad_signature_length' | 'duplicate_inputs' | 'input_missing_for_signature' | 'malformed_transaction' | 'malformed_transaction_rlp' | 'no_inputs' | 'signature_corrupt' | 'signature_missing_for_input'}
apps/omg_api/lib/api.ex:43: The pattern {'ok', _recovered_tx@1} can never match the type {'error','bad_signature_length' | 'duplicate_inputs' | 'input_missing_for_signature' | 'malformed_transaction' | 'malformed_transaction_rlp' | 'no_inputs' | 'signature_corrupt' | 'signature_missing_for_input'}
apps/omg_jsonrpc/lib/application.ex:20: Function start/2 has no local return
apps/omg_watcher/lib/web/controllers/challenge.ex:42: The pattern <{'ok', _challenge@1}, _conn@1> can never match the type <{'error','invalid_challenge_of_exit'},_>
```

**Pull request merged**